### ISSUE TITLE
[@typescript-eslint/indent] SwitchCase default inconsistent with docs and ESLint

### ISSUE THREAD

**Issue by benquarmby** [typescript-eslint/typescript-eslint#608]
The documentation states that the default indentation level for `SwitchCase` is zero:
https://github.com/typescript-eslint/typescript-eslint/blame/61e6385c65b147a5a8b4caa2f3241e5a92c9b1bf/packages/eslint-plugin/docs/rules/indent.md#L77

The default for ESLint is also zero:
https://eslint.org/docs/rules/indent#options

In practice, the level is actually `1`. Other than the documentation being incorrect, I'd argue that turning ESLint's `indent` off and setting `@typescript-eslint/indent` to `error` should not produce new violations on code covered by both rules.

**Repro**

```JSON
{
  "rules": {
    "indent": "off",
    "@typescript-eslint/indent": "error"
  }
}
```

```TS
switch (typeof true) {
case "string":
    throw new Error("never");
}
```

**Expected Result**
Does not raise a violation.

**Actual Result**
Raises a violation.

**Additional Info**
Can be worked around by explicitly setting the `SwitchCase` level. Unfortunately this requires being explicit about the indentation depth also:
```JSON
{
  "rules": {
    "indent": "off",
    "@typescript-eslint/indent": ["error", 4, {
      "SwitchCase": 0
    }]
  }
}
```
**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.10.2` |
| `@typescript-eslint/parser`        | `1.10.2` |
| `TypeScript`                       | `3.3.1` |
| `ESLint`                           | `5.12.1` |
| `node`                             | `10.15.4` |
| `yarn`                              | `1.16.0` |

**Comment by bradzacher** [comment]
Yeah the documentation is copy pasted from eslint core.
We don't do that any more for reasons like this, but haven't gone back to fix the older docs.

We set new defaults based off of typescript standard:
https://github.com/typescript-eslint/typescript-eslint/blob/61e6385c65b147a5a8b4caa2f3241e5a92c9b1bf/packages/eslint-plugin/src/rules/indent.ts#L98-L108

So we need to update the docs to match.

**Comment by benquarmby** [comment]
> We set new defaults based off of typescript standard

Got it. Is that standard inferred from how TypeScript is used by the TS team and their documentation?

**Comment by bradzacher** [comment]
In part, yes.

The recommended set and the defaults for rules were based off of three things: what the TS team does in the documentation, what community eslint configs setup when including the old `eslint-plugin-typescript`, and what I thought was the correct thing at the time.

For indentation it was very much the former - TS playground + TS documentation both use 4 space indentation with switch cases indented at 1.

**Comment by bradzacher** [comment]
Merging into #1824
